### ISSUE TITLE
Find a way to include evidences for annotations or properties in general

### ISSUE THREAD

**Issue by ljgarcia** [BioSchemas/specifications#141]
Sometimes we want to add an evidence to a property. For instance, we know a protein is associated to a disease or process or function based on some experiment, publication, expert opinion, etc. We want to be able to add this evidence.

@profgiuseppe you had a case for this, related to GO terms as far as I remember. Could you please add that example here so we have more information for this issue?

**Comment by profgiuseppe** [comment]
@ljgarcia in our test we used the creationMethod of the ProteinAnnotation profile.
Something like
```json
{
	"@type": "ProteinAnnotation",
	"identifier": "http://identifiers.org/GO:0012505", 
	"location": "C:endomembrane system", 
	"additionalType": "http://edamontology.org/operation_2489", 
	"creation_method": {
	    "@type": "PropertyValue", 
	    "additionalType": "http://purl.obolibrary.org/obo/ECO_0000217", 
	    "value": "http://purl.obolibrary.org/obo/ECO_0000256"
	}, 
	"preferredLabel": "ProteinAnnotation", 
	"additionalProperty": {
	    "url": "http://ourserver.biocomp.unibo.it", 
	    "additionalType": "http://edamontology.org/data_0007", 
	    "@type": "PropertyValue", 
	    "value": "ourserver"
	}
}
```

In our hypotetical usage, we had 

- identifier for the annotation
- creationMethod for the evidence code (ECO), this could be re-used in UniProt
- an additional property for the source of information (in this case, EDAM:data_0007 for "tool", but it could better be a Bioschemas tool profile pointing to the actual Bioschemas annotation of the tool)

**Comment by ljgarcia** [comment]
@profgiuseppe this looks fine to me and the creationMethod (all together, no underscore) exists for ProteinAnnotation. Now, one question, how do you relate a protein with this protein annotation? Would you use "contains" or something else? Would you use a ProteinAnnotation to model a relation to a GO molecular function or a GO biological process? If so, how would you link the protein to the protein annotation?

Also, I see this annotation is about a cellular location. Would it be right to say that a protein is contained by a cellular location?

**Comment by ljgarcia** [comment]
@profgiuseppe Would it make sense for you to use the following?
- Protein -> contains -> ProteinAnnotation linked to a GO subcellular location term? Of course if the main modeled entity is the ProteinAnnotation, then you would use ProteinAnnotation ->isContainedIn -> Protein
- Protein -> contains -> BioChemEntity representing a GO subcellular location if no evidence exists
- Protein -> contains -> ProteinAnnotation linked to, for instance, an organism if an evidence for such "contains" exist
- Protein -> contains -> BioChemEntity representing, for instance, an organism if no evidence exists
- Protein -> [ro:0002327](http://purl.obolibrary.org/obo/RO_0002327) -> ProteinAnnotation linked to a GO molecular function if an evidence exists
- Protein -> [ro:0002327](http://purl.obolibrary.org/obo/RO_0002327) -> BioChemEntity representing  a GO molecular function if no evidence exists
- Protein -> [ro:0002331](http://purl.obolibrary.org/obo/RO_0002331) ->ProteinAnnotation linked to a GO biological process if an evidence exists
- Protein -> [ro:0002331](http://purl.obolibrary.org/obo/RO_0002331) ->ProteinAnnotation representing a GO biological process if no evidence exists

In summary Protein -> property ->ProteinAnnotation if it is commonly considered a ProteinAnnotation or if an evidence is available and should be model. And Protein -> property -> BioChemEntity (any other than ProteinAnnotation) if no evidence is available.

**Comment by ljgarcia** [comment]
@Kerruba could you please provide here an example (or link to) on how you are using CategoryCode? I think we could use it here for ECO terms. Thanks!

**Comment by ljgarcia** [comment]
@profgiuseppe 

Please take a look to this example about a protein involved in a GO Biological process. We are using here the categoryCode property whenever there is a link to a well-known vocabulary for ProteinAnnotations or for evidences, i.e., creationMethod. Please notice that all the information regarding the CategoryCode (like the inCategoryCodeset) comes from schema.org and not Bioschemas so any field is always optional.
```
{
  "@type": ["BioChemEntity", "Protein"],
  "involvedIn": {
    "@type": "ProteinAnnotation",
    "identifier": "GO:0000278",
    "**categoryCode**": {
      "@type": "CategoryCode",
      "url": "http://purl.obolibrary.org/obo/GO_0000278",
      "codeValue": "GO:0000278",
      "name": "mitotic cell cycle"
    },
    "**creationMethod**": {
      "@type": "PropertyValue",
      "value": "imported manually asserted information used in automatic assertion",
      "valueReference": {
        "@type": "CategoryCode",
        "codeValue": "ECO:0000322",
        "inCategorySet": {
          "@type": "CategoryCodeSet",
          "name": "ECO Evidence Code Ontology"
        }
      }
    }
  }
}
```

**Comment by ljgarcia** [comment]
Moving it to a broader scope https://github.com/BioSchemas/specifications/issues/426
